src/ui/Omnibox/Omnibox.ts

```diff
@@ -692,6 +704,30 @@ export class Omnibox extends Component {
     new QueryboxQueryParameters(this.options).addParameters(data.queryBuilder, query);
   }
 
+  private handleNewQuery(data: INewQueryEventArgs) {
+    Assert.exists(data);
+    this.options.clearFiltersOnNewQuery && this.clearFiltersIfNewQuery(data);
+  }
+
+  private clearFiltersIfNewQuery({ origin, searchAsYouType }: INewQueryEventArgs) {
+    if (this.queryController.firstQuery) {
+      return;
+    }
+
+    // Prevent queries triggered by unrelated components to clear the the filters
+    // e.g., a facet selection
+    const validOrigins = [Omnibox.ID, 'SearchButton'];
```

Comment:
Should also include `Querybox` here.

Reply:
Well technically if you have a Querybox, you don't have an Omnibox, except if I go put that piece of code in the Querybox too or a level higher in the Searchbox 🤔

Reply:
@olamothe What do you think? Fine as it is? I don't think the Querybox is used that much I believe anymore... Most new clients will have the Searchbox/Omnibox package...